### ISSUE TITLE
Installer cannot verify php version - PHP 5.5.16-1+deb.sury.org~precise+1

### ISSUE THREAD

**Issue by mszterlingfsite** [magento/magento2#1066]
I'm on ubuntu and my php version is
```
PHP 5.5.16-1+deb.sury.org~precise+1 (cli) (built: Aug 25 2014 10:23:41) 
Copyright (c) 1997-2014 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2014 Zend Technologies
    with Zend OPcache v7.0.4-dev, Copyright (c) 1999-2014, by Zend Technologies
```

While using web installer - there is a ajax call to /setup/index.php/environment/php-version
and that is returning error 500

vendor/composer/composer/src/Composer/Package/Version/VersionParser.php:370

Could not parse version constraint 5.5.16-1+deb.sury.org~precise+1: Invalid version string "5.5.16-1+deb.sury.org~precise+1"

**Comment by mszterlingfsite** [comment]
Just to add comment - installed should be able to deal with error 500 returned by those ajax requests so message can be passed to the user.

**Comment by devgas** [comment]
I'm on ubntu too and have same issue. My php version is
"PHP 5.5.9-1ubuntu4.6".

For temporary solving issue I reverted changes to "MAGETWO-33385: Run static code tests in setup folder" commit.

**Comment by monkeysee** [comment]
Thanks for reporting, we have created internal ticket MAGETWO-34667 for the problem.

**Comment by mszterlingfsite** [comment]
If I can suggest - same as Seldaek - change the way it is done in Magento\Setup\Controller\Environment
Example how they parse php version in Composer is
https://github.com/composer/composer/blob/master/src/Composer/Repository/PlatformRepository.php#L38-44
```
        try {
            $prettyVersion = PHP_VERSION;
            $version = $versionParser->normalize($prettyVersion);
        } catch (\UnexpectedValueException $e) {
            $prettyVersion = preg_replace('#^([^~+-]+).*$#', '$1', PHP_VERSION);
            $version = $versionParser->normalize($prettyVersion);
        }
```

**Comment by versedi** [comment]
Same happening to me. 
```
UnexpectedValueException

File:
 /var/www/html/magento2/vendor/composer/composer/src/Composer/Package/Version/VersionParser.php:370

Message:
 Could not parse version constraint 5.5.12-2ubuntu4.2: Invalid version string "5.5.12-2ubuntu4.2"
```

**Comment by rorteg** [comment]
I agree with the @mszterlingfsite:
preg_replace('#^([^~+-]+).*$#', '$1', PHP_VERSION)

**Comment by safwkhan** [comment]
Thanks all for your suggestions as well! We are working on it (internal ticket# MAGETWO-34667).

**Comment by mazhalai** [comment]
This issue has been fixed in 0.42.0-beta11
Thank you for reporting the issue.
